pkg/controllers/provisioningv2/rke2/machineprovision/template.go

```diff
@@ -47,8 +47,22 @@ func (h *handler) objects(ready bool, typeMeta metav1.Type, meta metav1.Object,
 		return []runtime.Object{secret}
 	}
 
-	if args.BootstrapOptional && args.BootstrapSecretName == "" {
-		args.BootstrapSecretName = "not-found"
```

Comment:
Are there any possible adverse effects stemming from not setting this any more?

Reply:
No. The bootstrap secret is used to pass the system-agent install script. It is not used in the deletion scenario.